src/PricingModules/UniswapV3/UniswapV3PricingModule.sol

```diff
@@ -0,0 +1,362 @@
+
+        // Calculate amount0 and amount1 of the principal (the liquidity position without accumulated fees).
+        (amount0, amount1) = LiquidityAmounts.getAmountsForLiquidity(
+            sqrtPriceX96, TickMath.getSqrtRatioAtTick(tickLower), TickMath.getSqrtRatioAtTick(tickUpper), liquidity
+        );
+    }
+
+    /**
+     * @notice Calculates the sqrtPriceX96 (token1/token0) from trusted USD prices of both tokens.
+     * @param priceToken0 The price of 10^18 tokens of token0 in USD, with 18 decimals precision.
+     * @param priceToken1 The price of 10^18 tokens of token1 in USD, with 18 decimals precision.
+     * @return sqrtPriceX96 The square root of the price (token1/token0), with 96 binary precision.
+     * @dev The price in Uniswap V3 is defined as:
+     * price = amountToken1/amountToken0.
+     * The usdPriceToken is defined as: usdPriceToken = amountUsd/amountToken.
+     * => amountToken = amountUsd/usdPriceToken.
+     * Hence we can derive the Uniswap V3 price as:
+     * price = (amountUsd/usdPriceToken1)/(amountUsd/usdPriceToken0) = usdPriceToken0/usdPriceToken1.
+     */
+    function _getSqrtPriceX96(uint256 priceToken0, uint256 priceToken1) internal pure returns (uint160 sqrtPriceX96) {
+        // Both priceTokens have 18 decimals precision and result of division should also have 18 decimals precision.
+        // -> multiply by 10**18
+        uint256 priceXd18 = priceToken0.mulDivDown(1e18, priceToken1);
+        // Square root of a number with 18 decimals precision has 9 decimals precision.
+        uint256 sqrtPriceXd9 = FixedPointMathLib.sqrt(priceXd18);
+
+        // Change sqrtPrice from a decimal fixed point number with 9 digits to a binary fixed point number with 96 digits.
+        // Unsafe cast: Cast will only overflow when priceToken0/priceToken1 >= 2¨^128.
+        sqrtPriceX96 = uint160((sqrtPriceXd9 << FixedPoint96.RESOLUTION) / 1e9);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                    RISK VARIABLES MANAGEMENT
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Sets the maximum exposure for an underlying asset.
+     * @param asset The contract address of the underlying asset.
+     * @param maxExposure The maximum protocol wide exposure to the underlying asset.
+     * @dev Can only be called by the Risk Manager, which can be different from the owner.
+     */
+    function setExposureOfAsset(address asset, uint256 maxExposure) public override {
+        // Authorization that only Risk Manager can set a new maxExposure is done in parent function.
+        super.setExposureOfAsset(asset, maxExposure);
+
+        // If the maximum exposure for an asset is set for the first time, check that the asset can be priced
+        // by the erc20PricingModule.
+        if (exposure[asset].exposure == 0) {
+            require(PricingModule(erc20PricingModule).inPricingModule(asset), "PMUV3_SEOA: Unknown asset");
```

Comment:
it could in theory be that the underlying asset's oracle is decommissioned or not working. Not sure if we should catch this as it introduces a lot of complexity for an onlyRiskmgr function

Reply:
What does it matter to set an exposure?
GetValue would return 0 in that case